app/assets/src/components/views/discovery/mapping/MapPlayground.jsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Marker } from "react-map-gl";
+import { get } from "lodash/fp";
+
+import BaseMap from "~/components/views/discovery/mapping/BaseMap";
+import CircleMarker from "~/components/views/discovery/mapping/CircleMarker";
+
+class MapPlayground extends React.Component {
+  state = {
+    locationsToSamples: {},
+    viewport: {}
+  };
+
+  componentDidMount() {
+    const { results } = this.props;
+
+    const locationsToSamples = Object.assign({}, this.state.locationsToSamples);
```

Comment:
This seems to be part of initializing state for the first time and is not DOM related. You should do this on the constructor (or better, from an initialization function called from the constructor)

Reply:
thnx